src/integration/pages/storyPage/crossPlatformTests.js

```diff
@@ -3,17 +3,28 @@ import { runCommonCrossPlatformTests } from '../../common';
 export default () => {
   runCommonCrossPlatformTests();
 
-  it('I can see an image with a caption', () => {
-    const imageEl = document.querySelector(
-      'main figure img, main figure amp-img',
-    );
-    expect(imageEl).toBeInTheDocument();
+  const noscripts = document.querySelectorAll('noscript');
+  const firstImageLazyloaded = noscripts.length > 1;
 
-    const imageCaptionEl = document.querySelector('main figure figcaption');
-    expect(imageCaptionEl).toBeInTheDocument();
-    expect(imageCaptionEl.textContent).toBeTruthy();
-    expect(imageCaptionEl.textContent).toMatchSnapshot();
-  });
+  if (firstImageLazyloaded) {
```

Comment:
Why is the image sometimes lazy loaded, sometimes not; I'd expect the test behaviour to be consistent? Does the behaviour differ between canonical and amp?

Reply:
Yes the behaviour is different between canonical and amp. Amp prefetches lazy loaded resources so I don't think the images in amp use the `react-lazyload` library which canonical uses. https://amp.dev/about/how-amp-works/#prioritize-resource-loading

Reply:
In that case, I'd suggest we create seperate tests in `ampTests.js` and `canonicalTests.js` so it's clear why the behvaiour differs.